bcgov/nr-spar-webapp
# fix: update seedlot existing pages

# Description
Jira ticket: [FSADT2-547](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-547)

This PR consists mostly visual updates and some components changes.
Shout-out to @mgaseta for his help!

### Notable Changes

- All "Exisiting seedlot" files and class names have been updated to "My seedlots" for consistency
- the `EmptySection` component can now accept either a `Icon` or `Pictogram`. If both props are provided, `Pictogram` will be used
- `frontend/src/utils/filterUtils.ts` has been added, it's useful for Carbon input filtering
- the `ComboButton` component has an optional `menuOptionsClass` prop for style fine tuning

### How to test

1. Go to http://localhost:3000/seedlot
    - The title for the third card and the table should be "My seedlots" instead of "Existing seedlot" 
![image](https://user-images.githubusercontent.com/25162561/223853749-83d001ab-6652-4941-b37d-67200c140759.png)

2. Go to `frontend/src/components/SeedlotActivities/index.tsx`
    - on line 118, replace `cards.map` with `empty_cards.map`
    - on the same web page, the "My seedlot" and "Activity history" cards should look like this: 
![image](https://user-images.githubusercontent.com/25162561/223847874-7f20d094-068f-422c-83c1-2bee8b007da3.png)
3. On the same web page, click on the "Register an A class seedlot" card
    - the "Applicant agency name" and "Applicant agency number"'s helper text are updated
    - the "Applicant agency number" input is now a `ComboBox`, you can test the filter function by typing
        - note that the `ComboBox` will have a 'x' sign to clear the current field, which is inconsistent with the design, hopefully that's ok with the design team
4. Fill in the necessary fields and click on the "Create seedlot number" button
    - you should be redirected to the success page, and the success message should have the seedlot number we just generated 
![image](https://user-images.githubusercontent.com/25162561/223850681-c1760e2d-8fb6-44aa-80bd-89ccf3861c21.png)
5. Click on the "Go to seedlot's detail screen" button
    - The blue combo button on the right should look like this: 
![image](https://user-images.githubusercontent.com/25162561/223851380-88042260-9f02-4e98-87ac-3c7e77403279.png)
6. Go to `frontend/src/components/FormReview/index.tsx`
    - on line 65, replace `mockFormData.length` with `emptyMockFormData.length`
    - on the same web page, the "Form Review" section should look like this: 
![image](https://user-images.githubusercontent.com/25162561/223852263-cbe55595-c10e-4e52-88a0-998c8eebc576.png)



## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?
Tested locally and ran `yarn run lint`


## Checklist






---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-webapp-70-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-webapp/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-webapp-70-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-webapp/actions/workflows/merge-main.yml)
